pkg/operator/ceph/object/rgw.go

```diff
@@ -205,6 +205,12 @@ func (c *clusterConfig) startRGWPods() error {
 			if err != nil {
 				logger.Infof("failed to delete rgw key %q. %v", depNameToRemove, err)
 			}
+
+			// Delete configuration in centralized mon database
+			err = c.deleteFlagsMonConfigStore(depNameToRemove)
```

Comment:
While you're in here, there is some code duplication that needs to be cleaned up. Removing an rgw daemon is duplicated between this method (lines 189-213) and the `deleteStore()` method (lines 281-292). Looks like this method is more complete in its cleanup than `deleteStore()`. Could you factor these out to a common cleanup method so the rgw daemons are always cleaned up consistently? Thanks!

Reply:
I will factorize call to ceph command but for deletion of the deployment and secret key the `deleteStore()` method doesn't call it because it rely on kubernetes garbage collection mechanism which delete all object that have a owner reference to the cephobjectstore object being deleted.
We can perhaps link the secret to the deployment to avoid the secret code deletion but won't be able to do much more in the down scale code

Reply:
If we link the secret to the deployment it'll never be watched/reconciled if it gets modified/deleted. So I'd suggest keeping the current approach.

Reply:
True, the ownerReference is the reason they were different. Thanks for refactoring, it will be better to avoid any duplication even if the resource deletion is not needed in one of the code paths.